server/src/main/java/org/elasticsearch/index/store/StoreFileMetadata.java

```diff
@@ -33,17 +41,22 @@
 
     private final BytesRef hash;
 
+    private final BytesRef writerUuid;
+
     public StoreFileMetadata(String name, long length, String checksum, String writtenBy) {
```

Comment:
Should we get rid of this ctor somehow? It's used in `RecoveryFileChunkRequest` as a way to carry the name/length/etc but those are serialized separately there and I wonder if that could introduce some bugs later if someone rely on `writerUuid` in recovery but it's never available there.

Reply:
Good catch! Maybe we should serialize the `writerUuid` there too? It's a bit hacky but that's where we are today 🤔

Reply:
Instead of adding the serialization of the `writerUuid` we could maybe just serialize a `StoreFileMetadata`. This can be done in a follow up though